internal/clients/http/gcp/gcp_client.go

```diff
@@ -316,6 +316,7 @@ func (c *gcpClient) GetInstanceDescriptionByID(ctx context.Context, id, zone str
 	instanceId := strconv.FormatUint(instance.GetId(), 10)
 	instanceDesc := clients.InstanceDescription{ID: instanceId}
 	for _, n := range instance.NetworkInterfaces {
+		instanceDesc.PrivateIPv4 = *n.NetworkIP
```

Comment:
Similarly, I would perhaps check for nil unless there is a documentation remark somewhere that this is never unallocated pointer.